test_runner/regress/test_remote_storage.py

```diff
@@ -97,6 +98,15 @@ def test_remote_storage_backup_and_restore(
         pageserver_http.timeline_checkpoint(tenant_id, timeline_id)
 
         log.info(f"waiting for checkpoint {checkpoint_number} upload")
+
+        # insert upload failpoints to exercise retry code path
+        action = "3*return->off" if checkpoint_number == 0 else "off"
```

Comment:
The checkpoints are numbered 1 and 2, see line 82 above:
```
checkpoint_numbers = range(1, 3)
```
Therefore, I believe this `if checkpoint_number == 0` condition is never true.

Also, isn't it too late to set the failpoint, after the checkpoint has already been started?

I'd suggest moving this configure_failpoints call to before the loop

Reply:
Added a commit with that change to this PR